receiver/chronyreceiver/internal/chrony/client.go

```diff
@@ -75,13 +77,19 @@ func New(addr string, timeout time.Duration, opts ...clientOption) (Client, erro
 }
 
 func (c *client) GetTrackingData(ctx context.Context) (*Tracking, error) {
-	sock, err := c.dialer(c.proto, c.addr)
+	clk := clock.FromContext(ctx)
+
+	deadline := clk.Now().Add(c.timeout)
+
+	ctx, cancel := context.WithDeadline(ctx, deadline)
+	defer cancel()
```

Comment:
You can replace this to being:

```suggestion
   ctx, cancel := clock.TimeoutContext(ctx, c.timeout)
   defer cancel()
```

That way you don't need to apply math to `now`, it should all be handled for you.

Reply:
Sounds good! I still need `deadline` for `sock.SetDeadline` but I guess this way is a little cleaner